src/screens/PinScreen.js

```diff
@@ -195,9 +199,17 @@ class PinScreen extends React.Component {
 
   goToReset = () => {
     // navigate to reset screen
-    this.props.navigation.navigate('ResetWallet', { onBackPress: () => this.backFromReset() });
+    this.props.navigation.navigate('ResetWallet', {
+      onBackPress: () => {
+        this.props.startWalletLock();
+        this.backFromReset();
+      },
+    });
     // make sure we won't show loadHistory screen
     this.props.setLoadHistoryStatus(false, false);
+    // Start wallet might be set to LOADING, we also need to
+    // set it to READY so the loading screen will hide
+    this.props.startWalletSuccess();
```

Comment:
You set the wallet as ready even before the unlock only to show the reset wallet?

Reply:
😳

Refactored with a better solution:
3beb1d5ebe85f74cdbb28a05907623991cceeaa2